barretenberg/cpp/src/barretenberg/vm/generated/tests/AvmMini_arithmetic.test.cpp

```diff
@@ -0,0 +1,492 @@
+#include "barretenberg/ecc/curves/bn254/fr.hpp"
+#include "barretenberg/flavor/generated/AvmMini_flavor.hpp"
+#include "barretenberg/numeric/uint256/uint256.hpp"
+#include "barretenberg/proof_system/circuit_builder/AvmMini_helper.hpp"
+#include "barretenberg/proof_system/circuit_builder/AvmMini_trace.hpp"
+#include "barretenberg/sumcheck/sumcheck_round.hpp"
+#include "barretenberg/vm/generated/AvmMini_composer.hpp"
+#include "barretenberg/vm/generated/AvmMini_prover.hpp"
+#include "barretenberg/vm/generated/AvmMini_verifier.hpp"
+
+#include <cstddef>
+#include <cstdint>
+#include <gtest/gtest.h>
+#include <string>
+#include <vector>
+
+using namespace proof_system;
+
+namespace tests_avm {
+
+class AvmMiniArithmeticTests : public ::testing::Test {
+  public:
+    AvmMiniTraceBuilder trace_builder;
+
+  protected:
+    // TODO(640): The Standard Honk on Grumpkin test suite fails unless the SRS is initialised for every test.
+    void SetUp() override
+    {
+        barretenberg::srs::init_crs_factory("../srs_db/ignition");
+        trace_builder = AvmMiniTraceBuilder(); // Clean instance for every run.
+    };
+};
+
+class AvmMiniArithmeticNegativeTests : public AvmMiniArithmeticTests {};
+
+// We add some helper functions in the anonymous namespace.
+namespace {
+
+/**
+ * @brief Helper routine proving and verifying a proof based on the supplied trace
+ *
+ * @param trace The execution trace
+ */
+void validateTraceProof(std::vector<Row>&& trace)
+{
+    auto circuit_builder = AvmMiniCircuitBuilder();
+    circuit_builder.set_trace(std::move(trace));
+
+    EXPECT_TRUE(circuit_builder.check_circuit());
+
+    auto composer = honk::AvmMiniComposer();
+    auto prover = composer.create_prover(circuit_builder);
+    auto proof = prover.construct_proof();
+
+    auto verifier = composer.create_verifier(circuit_builder);
+    bool verified = verifier.verify_proof(proof);
+
+    if (!verified) {
+        log_avmMini_trace(circuit_builder.rows, 0, 10);
```

Comment:
im guessing the 10 here is just arbitrary, is it enough?

Reply:
Yeah it is arbitrary. I did not want to have a too high numbers. Otherwise, when it is failing the CI logs will contain huge traces which is not nice. The idea is that when you debug, you can adapt this constant.
I would keep it like that unless you insist.